clients/src/main/java/org/apache/kafka/common/utils/LogContext.java

```diff
@@ -40,20 +42,420 @@ public LogContext() {
+        @Override
+        public void trace(Marker marker, String msg) {
+            if (logger.isTraceEnabled()) {
+                logger.log(marker, fqcn, LocationAwareLogger.TRACE_INT, addPrefix(msg), null, null);
+            }
+        }
+
+        @Override
+        public void trace(Marker marker, String format, Object arg) {
+            if (logger.isTraceEnabled()) {
+                String formattedMsg = MessageFormatter.format(format, arg).getMessage();
+                logger.log(marker, fqcn, LocationAwareLogger.TRACE_INT, addPrefix(formattedMsg), new Object[]{arg}, null);
+            }
+        }
+
+        @Override
+        public void trace(Marker marker, String format, Object arg1, Object arg2) {
+            if (logger.isTraceEnabled()) {
+                String formattedMsg = MessageFormatter.format(format, arg1, arg2).getMessage();
+                logger.log(marker, fqcn, LocationAwareLogger.TRACE_INT, addPrefix(formattedMsg), new Object[]{arg1, arg2}, null);
+            }
+        }
+
+        @Override
+        public void trace(Marker marker, String format, Object... argArray) {
+            if (logger.isTraceEnabled()) {
+                String formattedMsg = MessageFormatter.arrayFormat(format, argArray).getMessage();
+                logger.log(marker, fqcn, LocationAwareLogger.TRACE_INT, addPrefix(formattedMsg), argArray, null);
+            }
+        }
+
+        @Override
+        public void trace(Marker marker, String msg, Throwable t) {
+            if (logger.isTraceEnabled()) {
+                logger.log(marker, fqcn, LocationAwareLogger.TRACE_INT, addPrefix(msg), null, t);
+
+            }
+        }
+
+        @Override
+        public void debug(String message) {
+            if (logger.isDebugEnabled()) {
+                logger.log(null, fqcn, LocationAwareLogger.DEBUG_INT, addPrefix(message), null, null);
+            }
+        }
+
+        @Override
+        public void debug(String format, Object arg) {
+            if (logger.isDebugEnabled()) {
+                final String formattedMsg = MessageFormatter.format(format, arg).getMessage();
+                logger.log(null, fqcn, LocationAwareLogger.DEBUG_INT, addPrefix(formattedMsg), new Object[]{arg}, null);
+            }
+        }
+
+        @Override
+        public void debug(String format, Object arg1, Object arg2) {
+            if (logger.isDebugEnabled()) {
+                final String formattedMsg = MessageFormatter.format(format, arg1, arg2).getMessage();
+                logger.log(null, fqcn, LocationAwareLogger.DEBUG_INT, addPrefix(formattedMsg), new Object[]{arg1, arg2}, null);
+            }
+        }
+
+        @Override
+        public void debug(String format, Object... args) {
+            if (logger.isDebugEnabled()) {
+                String formattedMsg = MessageFormatter.arrayFormat(format, args).getMessage();
+                logger.log(null, fqcn, LocationAwareLogger.DEBUG_INT, addPrefix(formattedMsg), args, null);
```

Comment:
Since we have to build the array anyway, we could add a helper to reduce some of the common boilerplate. Something like this for example:
```java
public void writeLog(marker, level, format, args, exception) {
  String formattedMsg = MessageFormatter.arrayFormat(format, args).getMessage();
  logger.log(null, fqcn, LocationAwareLogger.DEBUG_INT, addPrefix(formattedMsg), args, exception);
}
```
Then our implementations looks like this:
```java
public void debug(String format, Object arg) {
  if (logger.isDebugEnabled())
    writeLog(null, LocationAwareLogger.DEBUG_INT, format, new Object[]{arg}, null);
}
```
Think it's worth it?